app/src/webkit/java/org/mozilla/focus/webkit/TrackingProtectionWebViewClient.java

```diff
@@ -116,6 +115,32 @@ public void onReceivedError(final WebView webView, int errorCode,
         // we just check whether the failing URL is the current URL, which is enough to detect an error
         // in the main frame.
 
+        // WebView swallows odd pages and only sends an error (i.e. it doesn't go through the usual
+        // shouldOverrideUrlLoading), so we need to handle special pages here:
+        // about: urls are even more odd: webview doesn't tell us _anything_, hence the use of
+        // a different prefix:
+        if (failingUrl.startsWith("error:")) {
+            // format: error:<error_code>
+            final int errorCodePosition = "error:".length();
```

Comment:
I guess we could hardcode "6" here? Or even nicer: Move "error" to a public static final field and use that in both cases.